chain/beacon/node.go

```diff
@@ -262,10 +237,8 @@ func (h *Handler) run(startTime int64) {
 				// network conditions allow for it.
 				// XXX find a way to start the catchup as soon as the runsync is
 				// done. Not critical but leads to faster network recovery.
-				h.l.Debug("beacon_loop", "run_sync", "potential_catchup")
-				// XXX Find a way to not run sync again before another has
-				// finished - maybe merge with chain sync mechanism
-				go h.chain.RunSync(context.Background())
+				h.l.Debug("beacon_loop", "run_sync_catchup", "last_is", lastBeacon, "should_be", current.round)
+				go h.chain.RunSync(context.Background(), current.round, nil)
```

Comment:
should this either block or be a context that can be canceled? it seems very possible that we end up with multiple of these running in parallel?  (e.g `catchup` above calls `run` and then also runs this RunSync, and this is in a loop that could select multiple times potentially, no?

Reply:
The syncer doesnt starta new sync if it's already doing it so this call will return quick without doing the sync if it's already in progress.